openedx/features/courseware/tests.py

```diff
@@ -0,0 +1,203 @@
+                    "weight": 0.0
+                },
+                {
+                    "type": constants.POST_ASSESSMENT_FORMAT,
+                    "min_count": 1,
+                    "drop_count": 0,
+                    "short_label": constants.POST_ASSESSMENT_FORMAT,
+                    "weight": 0.25
+                }
+            ]
+        }
+
+        self.course.grading_policy = grading_policy
+        self.update_course(self.course, self.user.id)
+        self.course = self.store.get_course(self.course.id)
+
+
+class FetchPrePostAssessmentScoreTestCase(FetchPrePostAssessmentMixin):
+    """Check fetch pre post assessment scores properly"""
+
+    def setUp(self):
+        super(FetchPrePostAssessmentScoreTestCase, self).setUp()
+        self.basic_setup()
+
+    def test_unauthorized_request(self):
+        """Test un authorized request"""
+        chapter = ItemFactory.create(
+            parent_location=self.course.location,
+            category='chapter',
+            display_name='Welcome'
+        )
+
+        score_url = self.get_url(chapter)
+        response = self.client.get(score_url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class FetchPrePostAssessmentScoreForUnEnrolledUserTestCase(FetchPrePostAssessmentMixin):
+    """Check fetch pre post assessment scores properly"""
+
+    def setUp(self):
+        super(FetchPrePostAssessmentScoreForUnEnrolledUserTestCase, self).setUp()
+        self.basic_setup()
+        self.client.login(username=self.user.username, password=self.user.password)
+        self.client.force_authenticate(self.user)
+
+    def test_un_enrolled_user_request(self):
+        """Test unenrolled user request"""
+        chapter = ItemFactory.create(
+            parent_location=self.course.location,
+            category='chapter',
+            display_name='Welcome'
+        )
+
+        score_url = self.get_url(chapter)
+        response = self.client.get(score_url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class FetchPrePostAssessmentScoreForEnrolledUserTestCase(FetchPrePostAssessmentMixin):
+    """Check fetch pre post assessment scores properly"""
+
+    def setUp(self):
+        super(FetchPrePostAssessmentScoreForEnrolledUserTestCase, self).setUp()
+        self.basic_setup()
+        self.client.login(username=self.user.username, password=self.user.password)
+        self.client.force_authenticate(self.user)
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
+
+    def _add_pre_assessment_subsection(self, chapter):
+        """Add pre assessment in module/chapter"""
+        return ItemFactory.create(
+            parent_location=chapter.location,
+            display_name='Pre Assessment',
+            category='sequential',
+            metadata={'graded': True, 'format': constants.PRE_ASSESSMENT_FORMAT}
+        )
+
+    def _add_post_assessment_subsection(self, chapter):
```

Comment:
Lets make one method out of `_add_post_assessment_subsection` and `_add_post_assessment_subsection`